ui/admin/tests/acceptance/credential-store/credentials/delete-test.js

```diff
@@ -44,73 +46,183 @@ module(
+      assert.strictEqual(
+        getUsernamePasswordCredentialCount(),
+        usernamePasswordCredentialCount
+      );
     });
 
-    test('can accept delete credential via dialog', async function (assert) {
+    test('cannot delete a username & key pair credential without proper authorization', async function (assert) {
+      assert.expect(3);
+      const usernameKeyPairCredentialCount =
+        getUsernameKeyPairCredentialCount();
+      instances.usernameKeyPairCredential.authorized_actions =
+        instances.usernameKeyPairCredential.authorized_actions.filter(
+          (item) => item !== 'delete'
+        );
+      await visit(urls.usernameKeyPairCredential);
+      assert.strictEqual(currentURL(), urls.usernameKeyPairCredential);
+      assert.notOk(
+        find('.rose-layout-page-actions .rose-dropdown-button-danger')
+      );
+      assert.strictEqual(
+        getUsernamePasswordCredentialCount(),
+        usernameKeyPairCredentialCount
+      );
+    });
+
+    test('can accept delete username & password credential via dialog', async function (assert) {
       assert.expect(2);
       const confirmService = this.owner.lookup('service:confirm');
       confirmService.enabled = true;
       confirmService.confirm = sinon.fake.returns(resolve());
-      const credentialsCount = getCredentialsCount();
-      await visit(urls.credential);
+      const usernamePasswordCredentialCount =
+        getUsernamePasswordCredentialCount();
+      await visit(urls.usernamePasswordCredential);
       await click('.rose-layout-page-actions .rose-dropdown-button-danger');
-      assert.strictEqual(getCredentialsCount(), credentialsCount - 1);
+      assert.strictEqual(
+        getUsernamePasswordCredentialCount(),
+        usernamePasswordCredentialCount - 1
+      );
       assert.ok(confirmService.confirm.calledOnce);
     });
 
-    test('cannot cancel delete credential via dialog', async function (assert) {
+    test('can accept delete username & key pair credential via dialog', async function (assert) {
+      assert.expect(2);
+      const confirmService = this.owner.lookup('service:confirm');
+      confirmService.enabled = true;
+      confirmService.confirm = sinon.fake.returns(resolve());
+      const usernameKeyPairCredentialCount =
+        getUsernameKeyPairCredentialCount();
+      await visit(urls.usernameKeyPairCredential);
+      await click('.rose-layout-page-actions .rose-dropdown-button-danger');
+      assert.strictEqual(
+        getUsernameKeyPairCredentialCount(),
+        usernameKeyPairCredentialCount - 1
+      );
+      assert.ok(confirmService.confirm.calledOnce);
+    });
+
+    test('cannot cancel delete username & password credential via dialog', async function (assert) {
+      assert.expect(2);
+      const confirmService = this.owner.lookup('service:confirm');
+      confirmService.enabled = true;
+      confirmService.confirm = sinon.fake.returns(reject());
```

Comment:
I don't think this needs to be handled in this PR but should we perhaps not mock out the response and actually click through the dialog? Is there a a reason we need to do this?

Reply:
There seems to be a pattern of clicking through the dialog in the update-test to make sure it works and using mocks in the delete-test file. Not sure if this needs further discussion but I'm happy to change it to using try catch blocks to actually click through the dialog.

Reply:
Nvm, try catch blocks are not needed.